tests/unit/test_expression_tree/test_symbol.py

```diff
@@ -267,17 +267,17 @@ def test_has_spatial_derivatives(self):
         grad_div_eqn = pybamm.div(grad_eqn)
         algebraic_eqn = 2 * var + 3
         self.assertTrue(grad_eqn.has_spatial_derivatives())
-        self.assertTrue(grad_eqn.has_gradient())
-        self.assertFalse(grad_eqn.has_divergence())
+        self.assertTrue(grad_eqn.has_symbol_of_class(pybamm.Gradient))
```

Comment:
I think it would be okay to just rename the function `has` if there isn't any clashes. Then doing
```python
    grad_eqn.has(pybamm.Gradient)
```

This really minor though so I don't mind if you prefer to keep it the way it is. 
Is a little neater? Could also create a super class for Gradient and Divergence `pybamm.SpatialDerivative` (or something) and then can just use 
```
 grad_eqn.has(pybamm.SpatialDerivative)
```

Reply:
I prefer to err on the side of clarity on this one